ecosystem/platform/server/app/controllers/settings_controller.rb

```diff
@@ -27,19 +27,15 @@ def profile_update
 
   def connections
     store_location_for(current_user, request.path)
-    @authorizations = @user.authorizations.to_h do |authorization|
-      [authorization.provider, authorization]
-    end
+    @authorizations = @user.authorizations.group_by(&:provider)
   end
 
   def connections_delete
     store_location_for(current_user, request.path)
-    @authorizations = @user.authorizations.to_h do |authorization|
-      [authorization.provider, authorization]
-    end
+    @authorizations = @user.authorizations
 
-    provider = params.fetch(:authorization, {}).require(:provider)
-    authorization = @authorizations[provider]
+    auth_id = params.fetch(:authorization, {}).require(:id)
+    authorization = @authorizations.select { |auth| auth.id == auth_id }.first
```

Comment:
I guess we should include uid in the form data to select the right authorization.

Reply:
the auth_id is the pkey, so should be global unique